aten/src/ATen/DLConvertor.cpp

```diff
@@ -167,6 +168,32 @@ ScalarType toScalarType(const DLDataType& dtype) {
               "Unsupported kFloat bits " + c10::to_string(dtype.bits));
       }
       break;
+    case DLDataTypeCode::kDLBfloat:
+      switch (dtype.bits) {
+        case 16:
+          stype = ScalarType::BFloat16;
+          break;
+        default:
+          throw std::logic_error(
```

Comment:
As before, use the exception macros (here and wherever else an exception would be thrown)